fsmonitor.c

```diff
@@ -218,18 +230,20 @@ void refresh_fsmonitor(struct index_state *istate)
 	}
 	strbuf_release(&query_result);
 
-	/* Now that we've updated istate, save the last_update time */
-	istate->fsmonitor_last_update = last_update;
+	/* Now that we've updated istate, save the last_update_token */
+	istate->fsmonitor_last_update = strbuf_detach(&last_update_token, NULL);
```

Comment:
Maybe we should `free(istate->fsmonitor_last_update)` just before this statement, just in case that we already took ownership of a token previously and now need to release its memory?